stellar/contract/round/src/internal.rs

```diff
@@ -0,0 +1,802 @@
+
+        let new_approved_project = read_approved_projects(env);
+        log_update_approved_projects(env, round.id, new_approved_project);
+        extend_instance(env);
+    }
+
+    fn remove_approved_project(env: &Env, admin: Address, project_ids: Vec<u128>) {
+        admin.require_auth();
+
+        let round = read_round_info(env);
+
+        if round.owner != admin {
+            round
+                .admins
+                .iter()
+                .find(|round_admin| round_admin == &admin)
+                .expect("Only round owner or round admin can remove approved project");
+        }
+
+        let approved_project = read_approved_projects(env);
+        assert!(
+            project_ids.len() >= approved_project.len(),
+            "Can not remove (Out of bound)"
+        );
+
+        project_ids.iter().for_each(|project_id| {
+            let already_approved = is_project_approved(env, project_id);
+            assert!(already_approved, "Project not approved");
+        });
+
+        project_ids.iter().for_each(|project_id| {
+            remove_approved_project(env, project_id);
+        });
+
+        let new_approved_project = read_approved_projects(env);
+        log_update_approved_projects(env, round.id, new_approved_project);
+        extend_instance(env);
+    }
+
+    fn add_white_list(env: &Env, admin: Address, address: Address) {
+        admin.require_auth();
+
+        let round = read_round_info(env);
+
+        if round.owner != admin {
+            round
+                .admins
+                .iter()
+                .find(|round_admin| round_admin == &admin)
+                .expect("Only round owner or round admin can add white list");
+        }
+
+        let is_black_listed = is_black_listed(env, address.clone());
+        assert!(!is_black_listed, "Address is black listed");
+
+        let is_white_listed = is_white_listed(env, address.clone());
+        assert!(!is_white_listed, "Address already white listed");
+
+        add_to_white_list(env, address.clone());
+        extend_instance(env);
+        log_update_white_list(env, round.id, address, true);
+    }
+
+    fn remove_from_white_list(env: &Env, admin: Address, address: Address) {
+        admin.require_auth();
+
+        let round = read_round_info(env);
+
+        if round.owner != admin {
+            round
+                .admins
+                .iter()
+                .find(|round_admin| round_admin == &admin)
+                .expect("Only round owner or round admin can remove white list");
+        }
+
+        let is_white_listed = is_white_listed(env, address.clone());
+        assert!(is_white_listed, "Address is not white listed");
+
+        let is_black_listed = is_black_listed(env, address.clone());
+        assert!(!is_black_listed, "Address is black listed");
+
+        remove_from_white_list(env, address.clone());
+        extend_instance(env);
+        log_update_white_list(env, round.id, address, false);
+    }
```

Comment:
i'm not sure about whitelist, can we use the `list` contract?

Reply:
I can't find the whitelist setting on the design, not designed yet?

Reply:
same, i think it should be manually added since list also for registration and general usage